org.kryogenix.Pick.yml

```diff
@@ -0,0 +1,37 @@
+app-id: org.kryogenix.Pick
+runtime: org.gnome.Platform
+runtime-version: '42'
+sdk: org.gnome.Sdk
+command: pick-colour-picker
+
+finish-args:
+  - --share=ipc
+  - --socket=pulseaudio
+  - --socket=x11
+
+modules:
+  - name: pick-colour-picker
+    buildsystem: simple
+    build-commands:
+      - install -D pick-colour-picker-flatpak-runscript.sh /app/bin/pick-colour-picker
+      - chmod a+x /app/bin/pick-colour-picker
+      - cp -r pick /app/pick
+      - rm -rf /app/pick/__pycache__
+      - mkdir -p /app/share/metainfo/
+      - sed 's/pick-colour-picker.desktop/org.kryogenix.Pick.desktop/' pick-colour-picker.appdata.xml > /app/share/metainfo/org.kryogenix.Pick.appdata.xml
+      - install -D data/icons/scalable/apps/pick-colour-picker-symbolic.svg /app/share/icons/hicolor/scalable/apps/org.kryogenix.Pick.svg
+      - install -D data/icons/512x512/apps/pick-colour-picker.png /app/share/icons/hicolor/512x512/apps/org.kryogenix.Pick.png
+      - install -D data/icons/128x128/apps/pick-colour-picker.png /app/share/icons/hicolor/128x128/apps/org.kryogenix.Pick.png
+      - install -D data/icons/64x64/apps/pick-colour-picker.png /app/share/icons/hicolor/64x64/apps/org.kryogenix.Pick.png
+      - install -D data/icons/48x48/apps/pick-colour-picker.png /app/share/icons/hicolor/48x48/apps/org.kryogenix.Pick.png
+      - install -D data/icons/32x32/apps/pick-colour-picker.png /app/share/icons/hicolor/32x32/apps/org.kryogenix.Pick.png
+      - install -D data/icons/24x24/apps/pick-colour-picker.png /app/share/icons/hicolor/24x24/apps/org.kryogenix.Pick.png
+      - install -D data/icons/22x22/apps/pick-colour-picker.png /app/share/icons/hicolor/22x22/apps/org.kryogenix.Pick.png
+      - install -D data/icons/16x16/apps/pick-colour-picker.png /app/share/icons/hicolor/16x16/apps/org.kryogenix.Pick.png
+      - mkdir -p /app/share/applications/
+      - sed 's/Icon=.*/Icon=org.kryogenix.Pick/' pick-colour-picker.desktop > /app/share/applications/org.kryogenix.Pick.desktop
```

Comment:
there is `desktop-file-edit` to avoid having to use sed.

Also if you use `rename-icon` then you don't have do bother.

Similarly, just install the desktop file as is and use `rename-desktop-file`, this will also avoid having to do the sed command  line 21.

ie all of this can be made much shorter and simpler

Reply:
OK, I don't think I understand `rename-icon`.
I can use `rename-desktop-file` and `rename-appdata-file` OK, but rename-icon doesn't seem to work in my testing.

I have icons in the source tree in `data/icons/{sizes}/apps/pick-colour-picker.png`, which I copy into the flatpak build tree with `cp -r data/icons /app/share/icons`. So, at that point in the build process, `...share/icons/24x24/apps/pick-colour-picker.png` exists, as does `.../share/icons/48x48/apps/pick-colour-picker.png`, etc, etc; there are many copies of this icon at different sizes. Which value do I need to put in `rename-icon` to get it to rename these icons?

I have tried `rename-icon: pick-colour-picker.png` and `rename-icon: apps/pick-colour-picker.png` and neither work: both throw this error:

`Error: icon apps/pick-colour-picker.png not found below /path/to/.flatpak-builder/rofiles/rofiles-PhUUsC/files/share/icons`

so I'm a bit lost as to why that isn't working. If you have suggestions I'd be happy to hear them. (The flatpak docs don't explain a great deal about the rename- properties other than to name them :-))

Reply:
the documentation say to use the **icon name** which doesn't include the extension. Because you could have a .svg icon too (or only)

Reply:
the docs do say that. Unfortunately, it doesn't help; I tried "rename-icon: pick-colour-picker" first, and I get the same error. (This is the part I don't understand.)

`Error: icon pick-colour-picker not found below .../.flatpak-builder/rofiles/rofiles-KzpfEQ/files/share/icons`

Reply:
it only work if the icons are installed in the right place, aka in `${FLATPAK_DEST}/share/icons/hicolor/SIZE/apps/ICON_NAME`

Reply:
aha! `hicolor`! Thank you. Now it works. Incoming update to the PR to fix all these issues :-)

Reply:
I believe the below pushed update should resolve these issues, and thank you for the patience!